File: sbschoolmanager-api/src/main/java/com/sbschoolmanager/sbschoolmanager_api/model/Material.java
package com.sbschoolmanager.sbschoolmanager_api.model;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;
import lombok.NoArgsConstructor;
import java.math.BigDecimal;

@Entity
@Table(name ="MATERIAL")
@Getter
@Setter
@NoArgsConstructor
public class Material {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "Cod_Material")
    private Integer codMaterial;

    @ManyToOne
    @JoinColumn(name= "Cod_alumno_FK")
    private Alumno alumno;

    @Column(name = "Tipo_material")
    private String tipoMaterial;

    @Column(name = "Precio")
    private BigDecimal precio;

    @Column(name = "Tiempo")
    private Integer tiempo;
}

File: sbschoolmanager_app/lib/models/material.dart

import "alumno.dart";

class Material {

  final int? codMaterial;
  final Alumno alumno;
  final String tipoMaterial;
  final double precio;
  final int tiempo;

  Material({
    this.codMaterial,
    required this.alumno,
    required this.tipoMaterial,
    required this.precio,
    required this.tiempo,
  });

  factory Material.fromJson(Map<String, dynamic> json){
    return Material(
      codMaterial: json["codMaterial"],
      alumno: Alumno.fromJson(json["alumno"]),
      tipoMaterial: json["tipoMaterial"],
      precio: (json["precio"] as num).toDouble(),
      tiempo: json["tiempo"],
    );
  }

  Map<String, dynamic> toJson(){
    return{
      if(codMaterial != null)
      "codMaterial": codMaterial,
      "alumno": alumno.toJson(),
      "tipoMaterial": tipoMaterial,
      "precio": precio,
      "tiempo": tiempo,
    };
    
  }
}